source/type/message.cpp

```diff
@@ -442,6 +442,23 @@ bool Impact::is_valid(const boost::posix_time::ptime& publication_date,
     return false;
 }
 
+ActiveStatus Impact::get_active_status(const boost::posix_time::ptime& publication_date) const {
+    bool is_future = false;
+    for (const auto& period : application_periods) {
+        if (period.contains(publication_date)) {
+            return ActiveStatus::active;
+        }
+        if (!period.is_null() && period.begin() >= publication_date) {
+            is_future = true;
```

Comment:
```suggestion
           return ActiveStatus::future;
```
No?

Reply:
We loop over all application_periods to check if there is currently active, I don't think application_periods are ordered ?

Reply:
OK